File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
# def solution(participant, completion):
#     answer = ''
#     dic = {}
#     for i in set(participant):
#         dic[i] = 0
#     for i in participant:
#         dic[i] += 1
#     for i in completion:
#         dic[i] -= 1
#     for i in dic:
#         if dic[i] != 0:
#             answer += i
#     return answer

from collections import Counter

def solution(participant, completion):
    answer = list(Counter(participant)-Counter(completion))
    return answer[0]

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        HashMap<String, Integer> comp = new HashMap<>();
        for (String com : completion) {
            comp.put(com, comp.getOrDefault(com, 0)+1);
        }
        for (String par : participant) {
            int cnt = comp.getOrDefault(par, 0);
            comp.put(par, cnt-1);
            if (comp.get(par) == -1) return par;
        }
        return "";
    }
}